### PR TITLE
add subarch-select

### PR DESCRIPTION
**Pull request by mr-c**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- ~~If static libraries are linked in, the license of the static library is packaged.~~
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by mr-c** [comment]
@conda-forge/help-c-cpp ready for review!

File: recipes/subarch-select/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    make CPPFLAGS="-DNDEBUG -D_FORTIFY_SOURCE=2 -O2 -isystem $BUILD_PREFIX/include"
+    mkdir -p ${PREFIX}/bin
+    make install prefix=${PREFIX}
+  number: 0
+
+requirements:
+  build:
+    - {{ compiler('cxx') }}
+    - make
+    - cpu_features >=0.8
```
Comment (xhochy) [inline recipes/subarch-select/meta.yaml:29]: This should probably be in host

File: recipes/subarch-select/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+source:
+  - url: https://github.com/ekg/{{ name }}/archive/refs/tags/v{{ version }}.tar.gz
+    sha256: acb41a936637d7488997646f61f386f1869bd91aa093e0f48d6299c95c9f9851
+    patches:
+      - makefile_tweaks.patch
+
+build:
+  skip: True  # [win or osx]
+  script: |
+    make CPPFLAGS="-DNDEBUG -D_FORTIFY_SOURCE=2 -O2 -isystem $BUILD_PREFIX/include"
```
Comment (xhochy) [inline recipes/subarch-select/meta.yaml:20]: Probably this is better placed in a build.sh. You should be passing in here CPPFLAGS=$CPPFLAGS, and do the same for CXXFLAGS.

**Review by xhochy — CHANGES_REQUESTED** [review]

File: recipes/subarch-select/meta.yaml (hunk 2 above)
Comment (mr-c) [reply recipes/subarch-select/meta.yaml]: I moved this to a simplified `build.sh`, thanks

File: recipes/subarch-select/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  skip: True  # [win or osx]
+  number: 0
+
+requirements:
+  build:
+    - {{ compiler('cxx') }}
+    - make
+  host:
+    - cpu_features >=0.8
+  #run:
```
Comment (carterbox) [inline recipes/subarch-select/meta.yaml:27]:
```suggestion
```

**Review by carterbox — CHANGES_REQUESTED** [review]
I don't see a comment from @milot-mirdita agreeing to maintain. Otherwise LGTM.

File: recipes/subarch-select/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+requirements:
+  build:
+    - {{ compiler('cxx') }}
+    - make
+  host:
+    - cpu_features >=0.8
+
+test:
+  commands:
+    - $PREFIX/bin/subarch-select sse /bin/echo -- SSE detected | grep "SSE detected"
```
Comment (carterbox) [inline recipes/subarch-select/meta.yaml:30]:
```suggestion
    - subarch-select sse /bin/echo -- SSE detected | grep "SSE detected"
```

**Comment by milot-mirdita** [comment]
We discussed this previously. I agree to co-maintaining :)

**Pull request merged**